src/common/SteelSheets.hpp

```diff
@@ -0,0 +1,122 @@
+    /// @brief Determine the name of the current active print sheet profile.
+    ///
+    /// @param[out] buffer Buffer to store the print sheet profile
+    /// @param[in] length Size of the given buffer.
+    /// @return Number of characters written to the buffer. Number will be
+    ///        always less than MAX_SHEET_NAME_LENGTH
+    static uint32_t ActiveSheetIndex();
+    ///////////////////////////////////////////////////////////////////////////////
+    /// @brief Reset the given print sheet profile to the uncalibrated state.
+    ///
+    /// In case the index of the given print sheet profile is bigger than the
+    /// MAX_SHEETS method return false.
+    /// @param[in] index Index of the sheet profile
+    /// @return True when the profile was reset, False othewise.
+    static bool ResetSheet(uint32_t index);
+    ///////////////////////////////////////////////////////////////////////////////
+    /// @brief Returns number of calibrated sheets
+    ///
+    /// Printer use print sheet profile on the index 0 as a default so the method
+    /// return always at least 1 calibrated profile.
+    /// @return Return the count of the calibrated print sheet profiles.
+    static uint32_t NumOfCalibrated();
+    ///////////////////////////////////////////////////////////////////////////////
+    /// @brief Determine the name of the current active print sheet profile.
+    ///
+    /// @param[out] buffer Buffer to store the print sheet profile
+    /// @param[in] length Size of the given buffer.
+    /// @return Number of characters written to the buffer. Number will be
+    ///        always less than MAX_SHEET_NAME_LENGTH
+    static uint32_t ActiveSheetName(char *, uint32_t len);
+    ///////////////////////////////////////////////////////////////////////////////
+    /// @brief Determine the name of the given print sheet profile.
+    ///
+    /// @param[in] index Index of the sheet profile
+    /// @param[out] buffer Buffer to store the print sheet profile
+    /// @param[in] length Size of the given buffer.
+    /// @return Number of characters written to the buffer. Number will be
+    ///        always less than MAX_SHEET_NAME_LENGTH
+    static uint32_t SheetName(uint32_t index, char *, uint32_t len);
+    ///////////////////////////////////////////////////////////////////////////////
+    /// @brief Rename the given print sheet profile.
+    ///
+    /// @param[in] index Index of the sheet profile
+    /// @param[in] buffer New name of the print sheet profile
+    /// @param[in] length Size of the given name.
+    /// @return Number of characters written to the buffer. Number will be
+    ///        always less than MAX_SHEET_NAME_LENGTH
+    static uint32_t RenameSheet(uint32_t index, char const *, uint32_t len);
+    ///////////////////////////////////////////////////////////////////////////////
+    /// @brief Sets offset to currently selected sheet.
+    ///
```

Comment:
It would be worth to note that infinite Z offset is converted to 0 and clamped between Z_OFFSET_MIN and Z_OFFSET_MAX.